File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Az_SubCommessa/Models/az-subcommessa-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";
import { CheckObjOn_Id_Number, CheckObjOn_Id_Text } from "../../../ModelsBase/check-obj";
import { Az_SubCommessaUser4EditModel } from "../../Az_SubCommessaUser/Models/az-subcommessa-user-model";
import { Az_SubCommessaAttivita4EditModel } from "../../Az_SubCommessaAttivita/Models/az-subcommessa-attivita-model";

export class Az_SubCommessaModel {
  public id: number = 0;
  public idAz_Commessa: number = 0;
  public descrizione?: string;
  public default: boolean = false;
}

export class Az_SubCommessa_GetAll_InModel {}

export class Az_SubCommessa_GetAll_OutModel extends ModelResult {
  public az_SubCommessa: Az_SubCommessaModel[] = [];
}

// Modello per l'edit (incorpora tutti i dati che compongono la sub commessa)
export class Az_SubCommessa_4EditModel extends Az_SubCommessaModel {
  public az_SubCommessaUser: Az_SubCommessaUser4EditModel[] = [];
  public az_SubCommessaAttivita: Az_SubCommessaAttivita4EditModel[] = [];
  public az_SubCommessaSediReparto: CheckObjOn_Id_Number[] = [];
}

export class Az_SubCommessa_GetAll_4Edit_InModel {
  public id: number = 0; // id della subcommessa
}

export class Az_SubCommessa_GetAll_4Edit_OutModel extends ModelResult {
  public az_SubCommessa: Az_SubCommessa_4EditModel[] = [];
}

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Az_SubCommessaService/Models/Az_SubCommessaModel.cs
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_SubCommessaAttivitaService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_SubCommessaSediRepartoService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_SubCommessaUserService.Models;
using nvxapp.server.service.ClientServer_Service.ModelsBase;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_SubCommessaService.Models
{
    public class Az_SubCommessaModel
    {
        public int Id { get; set; }
        public int IdAz_Commessa { get; set; }
        public string? Descrizione { get; set; }
        public Boolean Default { get; set; }
        public string Data { get; set; } = string.Empty;
        public string DataA { get; set; } = string.Empty;
    }

    public class Az_SubCommessa_GetAll_InModel { }

    public class Az_SubCommessa_GetAll_OutModel : ModelResult
    {
        public List<Az_SubCommessaModel> Az_SubCommessa { get; set; } = new List<Az_SubCommessaModel>();
    }




    //incorpora tutti i dati che compongono la sub commessa
    public class Az_SubCommessa_4EditModel : Az_SubCommessaModel
    {
        public List<Az_SubCommessaUser4EditModel> Az_SubCommessaUser { get; set; } = new List<Az_SubCommessaUser4EditModel>();
        public List<Az_SubCommessaAttivita4EditModel> Az_SubCommessaAttivita { get; set; } = new List<Az_SubCommessaAttivita4EditModel>();
        public List<Az_SubCommessaSediReparto4EditModel> Az_SubCommessaSediReparto { get; set; } = new List<Az_SubCommessaSediReparto4EditModel>();
    }


    public class Az_SubCommessa_GetAll_4Edit_InModel
    {
        public int Id { get; set; }  // id della commessa
    }

    public class Az_SubCommessa_GetAll_4Edit_OutModel : ModelResult
    {
        public List<Az_SubCommessa_4EditModel> Az_SubCommessa { get; set; } = new List<Az_SubCommessa_4EditModel>();
    }


    public class Az_SubCommessa_PutAll_4Edit_InModel
    {
        public int Id { get; set; }  // id della commessa
        public List<Az_SubCommessa_4EditModel> Az_SubCommessa { get; set; } = new List<Az_SubCommessa_4EditModel>();
    }

    public class Az_SubCommessa_PutAll_4Edit_OutModel : ModelResult
    {
        public int Id { get; set; }  // id della commessa
        public List<Az_SubCommessa_4EditModel> Az_SubCommessa { get; set; } = new List<Az_SubCommessa_4EditModel>();
    }

}
